libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -46,7 +46,6 @@
 #include <variant>
 
 using namespace std;
-using namespace ranges;
```

Comment:
I think @cameel and I argued about this, and I was on the position we should always use ranges with the complete explicit path. Though that was from an opinion perspective and was not backed up by this reasoning of clashes.

Reply:
I'd actually even go further and say it's a mistake that we even import `std` arguing along the lines of https://stackoverflow.com/a/1453605/2725593 - but yeah, we at least can't just import *everything* and hope that it works, because it won't.

Reply:
I personally do like the explicit path for `std` too, but that again only from an opinion perspective :)

Reply:
``namespace views = ranges::views;`` would also be relatively harmless - *but* we also need to consider that once we have C++20, there will also be ``std::views`` after which that won't work anymore.

Reply:
In this example I would have difficulty understanding where those functions are implemented or what is exactly going on in the background:
```
soltestAssert(
	(_expectedEdges | keys | to<set>()) == (_callGraphs | keys | to<set>()) &&
	(set_difference(_expectedCreatedContractSets | keys, _expectedEdges | keys)).empty(),
	"Contracts listed in expectations do not match contracts actually found in the source file or in other expectations."
);
```

Reply:
I don't really buy that argument because we're already doing much worse by having `using ...` with `std` and our own modules. I often do have difficulty tracking down where a class or even just an inherited function comes from in our codebase already. You probably do not notice it as much because you're much more familiar with it :) For these the new convention would be a big improvement in my opinion.

And for cases like `ranges` where the current convention is to use fully qualified names it's just as clear once you get used to it. The information is there, just not next to every use of the name. I think this is a very good balance between having to always qualify and never qualifying at all.

Reply:
No, I meant our general convention of doing stuff like `using namespace std`, `using namespace solidity::util`, etc.

Well, It's not such a big deal, I will do `ranges::views::stuff` from now on. Just I think we could do even better :)

Reply:
I prefer explicit imports because you do not accidentally create name conflicts and then you don't know what you're using. It also helps you seeing quite quickly what you use from an import. I do agree however that if you have to import, say, 30+ symbols, then importing a whole (sub-)namespace may make it more maintainable again.